src/main/java/io/cdap/plugin/scd2/SCD2Iterator.java

```diff
@@ -89,6 +95,69 @@ protected StructuredRecord computeNext() {
       }
     }
 
+    // get the next record that is available to compute the end time, it must be a record with same id
```

Comment:
This method becomes over 100 lines. Can we structure it into smaller ones?

Reply:
checkstyle for these projects is 120 characters a line

Reply:
A lot of lines are just comments, don't think break it down makes things easier to read